### PR TITLE
fix(fx-2894): fixes after authentication intent on novo pages

### PR DESCRIPTION
**Pull request by dzucconi**
Closes: [FX-2894](https://artsyproduct.atlassian.net/browse/FX-2894)

This is a re-implementation of [this bit of code](https://github.com/artsy/force/blob/49b6a9b180398ae5a574a14902600bcf14f617a6/src/desktop/components/main_layout/client.coffee#L15-L35), which no longer runs on new pages post-novo.

A little on the verbose side — we should really just have a single polymorphic follow mutation? — but should be straightforward.

### CODE REVIEW

File: src/v2/Utils/Hooks/useAuthIntent/index.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  }
+}
+
+/**
+ * If someone is logged out and clicks a follow or save button we create a cookie
+ * specifying what action they were trying to take.
+ *
+ * This hook is what checks that cookie and runs that action after they are authenticated.
+ */
+export const useAuthIntent = () => {
```
Comment (damassi) [inline src/v2/Utils/Hooks/useAuthIntent/index.ts:79]: This is more of a general often-comes-up thing, but there's a lot of minor filler above this function from which the file is named after. It'd be great if future devs could open this file and immediately see this line, versus having to search all around for a) context and b) this code comment.

Comment (dzucconi) [reply src/v2/Utils/Hooks/useAuthIntent/index.ts:79]: It's funny, I like to see the context first and like to see things defined above where they are used — I always find it a bit confusing to see a statement with it's definition buried at the bottom. I have been trying to shift my mindset more towards yours lately. We'll see 😅

**Review by iskounen — APPROVED** [review]
LGTM 👍 🎉

**Review by anandaroop — APPROVED** [review]
Makes sense, thanks 🙏🏽

**Pull request merged**